### DISCUSSION TITLE
Prefix utilization is 0% when the container prefix and active prefix are same

### DISCUSSION THREAD

**Discussion by nextgenkitty** [netbox-community/netbox#5834] [Q&A]
As a sample, I have a Container prefix with 10.200.0.0/24 and created the same 10.200.0.0/24 prefix with status Active.
I expect the container utilization to be 100% but it's showing 0%. This logic doesn't make sense to me because I allocate a container as a supernet and I just happen to use the full subnet as active on a production device, which is a legitimate logic. Can we change this behavior? Or am I not using the IPAM correctly? What else can I do then?

**Comment by candlerb** [comment]
Do you have `ENFORCE_GLOBAL_UNIQUE = False` in your configuration?  It sounds like you've created two duplicate prefixes.  A container prefix only counts sub-prefixes towards utilization.

See: https://github.com/netbox-community/netbox/blob/develop/netbox/ipam/models.py#L597-L615

I guess you could try extending the query:

```
            queryset = Prefix.objects.filter(
                prefix__net_contained=str(self.prefix),
                vrf=self.vrf
            )
```

so that it also matches any prefix with the same prefix/vrf and `status != PrefixStatusChoices.STATUS_CONTAINER`.  But in general, I'm not a fan of duplicates, and I have `ENFORCE_GLOBAL_UNIQUE = True`.

**Reply by nextgenkitty** [reply]
Thank you very much. I modified what you mentioned as below and it worked nicely. In my case I don't use NetBox as a single source of truth. There is another allocation tool owned by another team and I set the allocated prefixes as Container, then I divide those into smaller subnets to apply to my devices. Therefore, duplicates in my environment is unavoidable.

```
            queryset = Prefix.objects.filter(
                prefix__net_contained_or_equal=str(self.prefix),
                vrf=self.vrf
            ).exclude(pk=self.pk)
```
